google-cloud-spanner/lib/google/cloud/spanner/convert.rb

```diff
@@ -106,7 +106,12 @@ def object_to_grpc_value obj, field = nil
                 obj.rewind
                 content = obj.read.force_encoding("ASCII-8BIT")
                 encoded_content = Base64.strict_encode64(content)
-                Google::Protobuf::Value.new(string_value: encoded_content)
+                Google::Protobuf::Value.new string_value: encoded_content
+              elsif field == :PROTO || obj.class.respond_to?(:descriptor)
```

Comment:
I think it will be simpler to check the type against `Google::Protobuf::MessageExts` (i.e. add a case to the case statement) rather than checking the class for `:descriptor` here. You can still verify that `field == :PROTO` within the `when` clause, just like is done in the `Hash` case above.